src/Common/RWLock.cpp

```diff
@@ -287,7 +288,14 @@ void RWLockImpl::dropOwnerGroupAndPassOwnership(GroupsContainer::iterator group_
         /// Prepare next phase
         if (!readers_queue.empty())
         {
-            rdlock_owner = readers_queue.begin();
+            if (next && readers_queue.size() > 1)
+            {
+                rdlock_owner = std::next(readers_queue.begin());
+            }
+            else
+            {
+                rdlock_owner = readers_queue.begin();
+            }
```

Comment:
```suggestion
            rdlock_owner = readers_queue.begin();
            if (next && readers_queue.size() > 1)
                ++rdlock_owner;
```

Reply:
The purpose was to explicitly state the separate code path for  `next`, to avoid reader looking into other conditions and to show that `next` affects only readers queue.
Hence I prefer original variant here.

Although here I don't have strong objections and I can change it the suggested code.

Reply:
your argument makes sense, so let's left it as is